packages/data-context/src/sources/RelevantRunsDataSource.ts

```diff
@@ -243,11 +245,18 @@ export class RelevantRunsDataSource {
     if (!isEqual(toCache, this.#cached)) {
       debug('Values changed')
 
+      debug('current cache: %o, new values: %o', this.#cached, toCache)
+
       //TODO is the right thing to invalidate?  Can we just invalidate the runsByCommitShas field?
       const projectSlug = await this.ctx.project.projectId()
 
       await this.ctx.cloud.invalidate('Query', 'cloudProjectBySlug', { slug: projectSlug })
 
+      // If the cache is empty, then we're just starting up. Don't send notifications
+      if (this.#cached.all.length !== 0 && this.#cached.all[0] && toCache.all[0] && !isEqual(toCache.all[0], this.#cached.all[0])) {
```

Comment:
Do we want to only focus on the first run, or go through the cache and call the `maybeSendRunNotification`  for each pair of runs (cached vs new) that are not equal?  

An example scenario would be when multiple runs are kicked off for one commit.

Reply:
Also, I think because of the check for `this.#cached.all.length !== 0` that is there to prevent it from firing when the app first starts up, then i think this would not fire if the app had been open a while with no runs being fetched, and then a new run comes in.  This scenario is one I don't think needs too much time on and could be filed as a fast-follow in my opinion.  The users that will get the most value from this will be ones that already have runs.

Reply:
https://github.com/cypress-io/cypress/pull/26974/commits/95270660c9030c1a8b5677dbf22d16a2985ffa6e

`this.#cached.all[0]` implies `length !== 0`, so I removed that part.